Prohibit usage of retired names in semantic conventions (#2191)
open-telemetry/opentelemetry-specification: specification/common/attribute-naming.md

<!-- toc -->

- [Name Pluralization guidelines](#name-pluralization-guidelines)

> - [Name Reuse Prohibition](#name-reuse-prohibition)

Reviewer: Changelog please :-)

Author: Added.

  [Metric Name Pluralization Guidelines](../metrics/semantic_conventions/README.md#pluralization)
  SHOULD be followed for the attribute name.

## Name Reuse Prohibition

> It is prohibited to introduce a new attribute with a name that matches a name of

Reviewer 2: Can we use MUST NOT or SHOULD NOT language here instead of prohibited.

Reviewer 3: ```suggestion
A new attribute MUST NOT be added with the same name as
```

Author: Fixed.
